app/frontendapi.py

```diff
@@ -0,0 +1,112 @@
+from django.http import JsonResponse
```

Comment:
Optional: Can be done later, but if we expect each of the class here gets somewhat complex, may be better to split each class into its own file?

Reply:
My goal is to have basically no complexity in this file itself: ideally all the logic will be factored out and shared with the code for the Lite/AMP UI (and the API too, if needed), like with the search code.